### PR TITLE
Configuration file for command line switches

### PR DESCRIPTION
**Pull request by kris-nova**
Hey @justinsb - 

Here is a design proposal for the work highlighted in [issue 267](https://github.com/kubernetes/kops/issues/267).

Cheers

@chrislovecnm FYI

### CODE REVIEW

**Comment by kris-nova** [comment]
To elaborate on this further - the .yml configuration file will essentially be a YAML representation of [this struct ](https://github.com/kubernetes/kops/blob/master/cmd/kops/create_cluster.go#L17) which is essentially a representation of command line flags for `kops create cluster`.

The .yml configuration file should match the directives in the struct verbatim.

Cheers

File: docs/design/267.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Where `example.yml` would be a representation of a cluster spec.
+
+# Design
+
+#### create_cluster.go
+
+##### func (c *CreateClusterCmd) Run(args []string) error
+ 
+        err := rootCommand.ProcessArgs(args)
```
Comment (timothysc) [inline docs/design/267.md]: might want to consider viper as we're switching upstream main repo to use that.

**Comment by justinsb** [comment]
So this is great @kris-nova , but as @timothysc pointed out we're using viper which may do some of this for us: https://github.com/kubernetes/kops/blob/master/cmd/kops/root.go#L52

Can you maybe give that a go, and then let me know whether this is still needed (I expect it is, but maybe we can just integrate with viper somehow to avoid having to manually map every flag?)

**Comment by kris-nova** [comment]
Hey @justinsb - so that is exactly what I am doing in [PR 450](https://github.com/kubernetes/kops/pull/450) - which is actually the implementation piece of this work.

Looks like this design doc (obviously) is conflicting with the actual implementation in code. Let me update this PR with an updated design.

But if you look in 450 you will see that [I am using Viper](https://github.com/kubernetes/kops/pull/450/files#diff-e938618ffaa3cf1929fc0a2159e63fbeR11) (the pattern doesn't conflict with Root) and [then doing a passive](https://github.com/kubernetes/kops/pull/450/files#diff-7aee936424f26e427c2cfb8965c66326R62) `Unmarshal()` onto the Cobra command struct. So the values are completely dynamic IE: We aren't manually mapping every flag.

Is having two PRs the right way of doing this? I remember @chrislovecnm mentioning something about having the first PR with design and having a second PR with the implementation.

**Comment by kris-nova** [comment]
Closing - see comments on https://github.com/kubernetes/kops/pull/450.

**Pull request closed without merging**